SERVICES-662 user preferences e2e tests

harnash opened the pull request:
This integration tests check if User Preference Service is reachable from the Media Wiki side. It tries to get/set some user preferences on a test user account. It also disables user preference caching for some requests to see if the service stores properties properly. 

I had to make another PR since previous branch was merged into dev...
@Wikia/services-team

harnash committed:
Added integration test to UserPreferences.

harnash committed:
Added tests and implementation to bypass UserPreferences caching.

Caching inside UserPreferences class was hiding the problem that not
all cached values were properly saved into the storage. This change
allows test UserPreference without it's internal cache which exposes
a serious bug described in SERVICES-757.

harnash committed:
Merge branch 'dev' into SERVICES-662-user_preferences_e2e

harnash committed:
UserPreference integration tests now use "marketingallowed" property.

Since previously used property 'editor' is not a default one we should
rely on more common property which is set during registration.

harnash committed:
Merge remote-tracking branch 'origin/user-preferneces-service-integration' into SERVICES-662-user_preferences_e2e

harnash committed:
Merge branch 'user-preferneces-service-integration' into SERVICES-662-user_preferences_e2e

harnash committed:
Merge branch 'dev' into SERVICES-662-user_preferences_e2e

harnash committed:
Fixed integration test after merge.

PreferenceServiceImpl constructor changed and it broke the intergration
tests. Also some minor fixes were applied.

sqreek commented on lib/Wikia/tests/Service/User/UserPreferencesIntegrationTest.php:
```diff
@@ -0,0 +1,100 @@
+	/**
+	 * @var string
+	 */
+	protected $testUser;
```
testUserName?

harnash replied:
Fixed in ab2e8aa50ee884ea46f92c8658d69c3961aac275

nmonterroso commented on lib/Wikia/tests/Service/User/UserPreferencesIntegrationTest.php:
```diff
@@ -0,0 +1,100 @@
+		$this->testUserId = \User::idFromName( $this->testUser );
+		$this->preferenceServiceCached = Injector::getInjector()->get( PreferenceService::class );
+		$defaultPreferences = Injector::getInjector()->get( PreferenceServiceImpl::DEFAULT_PREFERENCES );
+		$cache = $this->getMockBuilder( CacheProvider::class )
```
There's also `VoidCache` which doesn't cache anything. Don't _need_ to change, but in the future it'd be nice to use that since it was made specifically for tests

harnash replied:
Thanks. Fixed that in 7c47bece3b688a31b239db846eff5d60c55ef955.

nmonterroso commented on lib/Wikia/tests/Service/User/UserPreferencesIntegrationTest.php:
```diff
@@ -0,0 +1,100 @@
+		$this->assertEquals( $testPref, 0, "Error resetting user preference [cached] '" . self::TEST_PREFERENCE_NAME . "' for user id: '$this->testUserId'" );
+	}
+
+	function testSetSingleUserPreferenceNotCached() {
```
Should we also have a test for getting without a cache?

harnash replied:
Added more tests in 58a03f509e944f251da1cc1c1885c413b822a0ca

nmonterroso commented:
one comment otw lgtm!

harnash committed:
Changed $testUser -> $testUserName in User Preference integration tests.

harnash committed:
Using VoidCach instead of mock in USer Preference integration tests.

harnash committed:
Added more integration tests for user preference service.

Added not cached get() test, missing preference test and check if cached
and not cached responses returns the same value.

harnash committed:
Merge branch 'dev' into SERVICES-662-user_preferences_e2e

nmonterroso commented:
LGTM - test failures seem unrelated so I'll merge
